### PR TITLE
LEAF839 Import Data Into New Form

### PR DESCRIPTION
**Pull request by jeanmarcablevets**
Added ability to make new a form and requests from an uploaded sheet.

Also added:

* activity/import progress indicator
* error checking: 
     * prevents columns with blank entries from being selected as required or in a required indicator
* generates report of requests made

### CODE REVIEW

File: LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl [hunk 1]
```diff
@@ -169,37 +270,54 @@
...
         var columnSelect = $(document.createElement('td'))
-            .append(buildSheetSelect(indicator.indicatorID, sheet_data))
+            .append(buildSheetSelect(indicator.indicatorID, sheet_data, indicator.required))
             .appendTo(row);
 
         return row;
     }
 
-    function doStuff(initiator, requestData) {
+    function doStuff(categoryID, initiator, requestData) {
```
Comment (jeremyrl) [inline LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl]: Rename this function to something that describes what it does.

File: libs/jsapi/portal/LEAFPortalAPI.js [hunk 2]
```diff
@@ -386,6 +386,99 @@ var PortalFormEditorAPI = function (baseAPIURL) {
                 .fail(onFail);
         },
 
+        /**
+         * Assigns workflow to form
+         *
+         * @param categoryID    string              name of form
+         * @param workflowID    int                 description of form
```
Comment (jeremyrl) [inline libs/jsapi/portal/LEAFPortalAPI.js]: These comments are incorrect.

File: LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl (hunk 1 above)
Comment (jeanmarcablevets) [reply LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl]: Addressed in commit `ddf`

File: libs/jsapi/portal/LEAFPortalAPI.js (hunk 2 above)
Comment (jeanmarcablevets) [reply libs/jsapi/portal/LEAFPortalAPI.js]: Addressed in commit `ddf`

**Pull request merged**